android/src/main/java/com/brentvatne/react/ReactVideoView.java

```diff
@@ -42,6 +44,26 @@
 
 import javax.annotation.Nullable;
 
+class CustomMediaController extends MediaController {
+    public CustomMediaController(Context context) {
+        super(context);
+    }
+
+    @Override
+    public boolean dispatchKeyEvent(KeyEvent event) {
+        int keyCode = event.getKeyCode();
+        if (keyCode == KeyEvent.KEYCODE_BACK || keyCode == KeyEvent.KEYCODE_MENU) {
+            Activity activity = ((ThemedReactContext) this.getContext()).getCurrentActivity();
+            if (activity != null) {
+                activity.onBackPressed();
+            }
+            return true;
```

Comment:
Should we return true only if we have an activity, or we need to in all cases where those keycodes are pressed? Can you also explain a little more why an onTouch event is dispatching a key event?

Reply:
When we are using the new `Predictive Back gesture` it triggers an onTouch event, since when we are pressing on the edges of the screen (it triggers an `onTouch` event) and when we pull right it should go back but since the controls are displayed it only hides the controls. This overrides that behavior so now onBack we go back no matter if the controls are displayed or not.

Reply:
> Should we return true only if we have an activity, or we need to in all cases where those keycodes are pressed
I think we should return true no matter if we have an activity or not. Most of the times we will have an activity, the check there is just to prevent app from crashing if we call `onBackPressed` but there is no activity.